src/types.rs

```diff
@@ -635,11 +654,21 @@ impl SourceMap {
         }
     }
 
+    /// Sets range_token indices.
+    ///
+    /// A token is a range token if the index is element of range_tokens.
+    pub fn set_range_tokens(&mut self, range_tokens: Vec<u32>) {
+        self.range_tokens = range_tokens;
+    }
+
     /// Looks up a token by its index.
     pub fn get_token(&self, idx: u32) -> Option<Token<'_>> {
-        self.tokens
-            .get(idx as usize)
-            .map(|raw| Token { raw, i: self, idx })
+        self.tokens.get(idx as usize).map(|raw| Token {
+            raw,
+            i: self,
+            idx,
+            is_range: self.range_tokens.binary_search(&idx).is_ok(),
```

Comment:
This is very inefficient. Could we make `range_tokens` a bitvec, to allow fast access and efficient memory usage?

Reply:
What do you think about storing a bool on the token itself when the sourcemap is parsed?

Reply:
I'll move the bool flag to the Token itself.